examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/XdsHelloWorldClient.java

```diff
@@ -0,0 +1,107 @@
+
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  /** Construct client for accessing HelloWorld server using the existing channel. */
+  public XdsHelloWorldClient(Channel channel) {
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+  }
+
+  /** Say hello to server. */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+    logger.info("Greeting: " + response.getMessage());
+  }
+
+  /**
+   * Greet server. If provided, the first element of {@code args} is the name to use in the
+   * greeting. The second argument is the target server. A {@code --secure} flag is also accepted.
+   */
+  public static void main(String[] args) throws Exception {
+    String user = "xds world";
+    // The example defaults to the same behavior as the hello world example. To enable xDS, pass an
+    // "xds:"-prefixed string as the target.
+    String target = "localhost:50051";
+    ChannelCredentials credentials = InsecureChannelCredentials.create();
+    if (args.length > 0) {
+      if ("--help".equals(args[0])) {
+        System.out.println("Usage: [--secure] [NAME [TARGET]]\n");
+        System.err.println("  --secure  Use credentials provided by xDS. Defaults to insecure");
+        System.err.println("  NAME      The name you wish to be greeted by. Defaults to " + user);
+        System.err.println("  TARGET    The server to connect to. Defaults to " + target);
+        System.exit(1);
+      } else if ("--secure".equals(args[0])) {
+        // The xDS credentials use the security configured by the xDS server when available. When
+        // xDS is not used or when xDS does not provide security configuration, the xDS credentials
+        // fall back to other credentials (in this case, InsecureChannelCredentials).
+        credentials = XdsChannelCredentials.create(InsecureChannelCredentials.create());
+        args = Arrays.copyOfRange(args, 1, args.length);
+      }
+    }
+    if (args.length > 0) {
+      user = args[0];
+    }
+    if (args.length > 1) {
+      target = args[1];
+    }
+
+    // This uses the new ChannelCredentials API. Grpc.newChannelBuilder() is the same as
+    // ManagedChannelBuilder.forTarget(), except that it is passed credentials. When using this API,
+    // you don't use methods like `managedChannelBuilder.usePlaintext()`, as that configuration is
+    // provided by the ChannelCredentials.
+    ManagedChannel channel = Grpc.newChannelBuilder(target, credentials)
+        .build();
+    try {
+      XdsHelloWorldClient client = new XdsHelloWorldClient(channel);
+      client.greet(user);
```

Comment:
Would you consider an enhancement to do this in a loop, so the client creates one channel but sends multiple RPCs on it? Could be useful in the xDS case with multiple backends etc. The loop count could be another argument.

It doesn't have to be in this PR but may be in future.

Reply:
I don't think we want to do that in _hello world_; it's supposed to be super-basic. There was work on another example, I think grpc coin, that would let you see load balancing working. For hello world, I think we'd just tell users to execute it more than once and see they get multiple backends. Yes, that doesn't show you get multiple backends with one channel, but if a user really cares they can just duplicate this line; hello world is intended to be super-easy to understand and edit so that users can toy with it in ways like that.